ethcore/src/engines/authority_round/mod.rs

```diff
@@ -97,13 +99,26 @@ const U16_MAX: usize = ::std::u16::MAX as usize;
 
 impl From<ethjson::spec::AuthorityRoundParams> for AuthorityRoundParams {
 	fn from(p: ethjson::spec::AuthorityRoundParams) -> Self {
-		let mut step_duration_usize: usize = p.step_duration.into();
-		if step_duration_usize > U16_MAX {
-			step_duration_usize = U16_MAX;
-			warn!(target: "engine", "step_duration is too high ({}), setting it to {}", step_duration_usize, U16_MAX);
-		}
+		let map_step_duration = |u: ethjson::uint::Uint| {
+			let mut step_duration_usize: usize = u.into();
+			if step_duration_usize == 0 {
+				panic!("AuthorityRoundParams: step duration cannot be 0");
+			}
+			if step_duration_usize > U16_MAX {
+				step_duration_usize = U16_MAX;
+				warn!(target: "engine", "step duration is too high ({}), setting it to {}", step_duration_usize, U16_MAX);
+			}
+			step_duration_usize as u16
+		};
+		let step_durations: BTreeMap<u64, u16> = match p.step_duration {
+			StepDuration::Single(u) => iter::once((0, map_step_duration(u))).collect(),
+			StepDuration::Transitions(tr) => {
```

Comment:
Do we verify there is at least one transition?

Reply:
In fact, we should verify that there is a transition for time `0`, i.e. an initial step duration.